web-app/src/components/BookingCalendar/container.js

```diff
@@ -36,21 +40,65 @@ const BookingCalendarContainer = Wrapped =>
       this.props.setEventBeingUpdated(isBeingUpdated);
     };
 
-    onSelectSlot = ({ start, end }) => {
+    checkIfPastDatesSelected = start => {
       const today = new moment();
-      if (moment(start).isAfter(today.add(-1, 'days'))) {
+      return moment(start).isBefore(today);
+    };
+
+    selectedDatesOverlapExisting = (start, end) => {
+      const overlappingEvents = this.props.takenHolidays.filter(hol => {
+        if (hol.employee && hol.employee.employeeId === this.props.employeeId) {
+          var selectedDateRange = moment.range(
+            moment(start),
+            moment(end).endOf('day'),
+          );
+          var existingEvent = moment.range(moment(hol.start), moment(hol.end));
+          if (selectedDateRange.overlaps(existingEvent)) {
+            return true;
+          }
+        }
+      });
+      return overlappingEvents.length > 0 ? true : false;
+    };
+
+    validateSelectedDates = (start, end) => {
+      const pastDatesSelected = this.checkIfPastDatesSelected(start);
+      if (pastDatesSelected) {
+        return [false, 'Unable to select past dates'];
+      } else {
+        const datesOverlapExisting = this.selectedDatesOverlapExisting(
+          start,
+          end,
+        );
+        if (datesOverlapExisting) {
+          return [
+            false,
+            'Your are trying to request dates that have already been set',
+          ];
+        } else {
+          return [true, 'dates approved'];
+        }
+      }
+    };
```

Comment:
Not sure we need the complexity of using an array as a return value here.

It could just return a error message if there's an error, and nothing if it's successful. Then below use `title: error || 'Dates approved'`,

Reply:
Updated to return a string. If string return equals 'Dates approved' it opens modal, otherwise displays error message

Reply:
Not a huge fan of this approach either. It feels brittle to have logic tied to what we're rendering. But I'll not get hung up in it.